src/services/post.ts

```diff
@@ -12,67 +15,153 @@ interface CreatePostData extends PostData {
+              { $lookup: { from: 'users', localField: 'user', foreignField: '_id', as: 'user' } },
+              { $unwind: '$user' },
+              { $match: tagQuery },
+              { $lookup: { from: 'tags', localField: 'tags', foreignField: '_id', as: 'tags' } },
+              { $sort: orderQuery },
+              { $skip: page > 0 ? (page - 1) * limit : 0 },
+              { $limit: limit },
+            ],
+          },
+        },
+      ]);
+    } catch (e) {
+      console.error(e);
+      throw e;
+    }
   }
 
   async getOnePost(data: { postId: string }) {
-    const { postId } = data;
-
-    const post = await Post.findOneAndUpdate(
-      {
-        _id: postId,
-      },
-      {
-        $inc: { viewsCount: 1 },
-      },
-      {
-        returnDocument: 'after',
-        lean: true,
-      },
-    );
-
-    return post;
+    try {
+      const { postId } = data;
+
+      const post = await Post.findOneAndUpdate(
+        {
+          _id: postId,
+        },
+        {
+          $inc: { viewsCount: 1 },
+        },
+        {
+          returnDocument: 'after',
+          lean: true,
+        },
+      );
+
+      return post;
+    } catch (e) {
+      console.error(e);
+      throw e;
+    }
   }
 
-  async updatePost(data: PatchPost) {
-    const { postId, ...otherData } = data;
-    const post = await Post.findById(postId);
-    if (!post) {
-      throw new createError.UnprocessableEntity();
+  async updatePost(data: PatchPost): Promise<UpdateResult> {
+    try {
+      const { postId, ...otherData } = data;
+      const post = await Post.findById(postId);
+      if (!post) {
+        throw new createError.UnprocessableEntity();
+      }
+
+      const updatedPost = await Post.updateOne(
+        {
+          _id: postId,
+        },
+        {
+          ...otherData,
+        },
+      ).lean();
+
+      return updatedPost;
+    } catch (e) {
+      console.error(e);
+      throw e;
     }
-
-    const updatedPost = await Post.updateOne(
-      {
-        _id: postId,
-      },
-      {
-        ...otherData,
-      },
-    ).lean();
-
-    return updatedPost;
   }
 
   async deletePost(data: { postId: string }) {
-    await Post.findOneAndDelete({
-      _id: data.postId,
-    }).lean();
+    try {
+      await Post.findOneAndDelete({
+        _id: data.postId,
+      }).lean();
+    } catch (e) {
+      console.error(e);
+      throw e;
+    }
+  }
+
+  async getUserPosts(userId: string, limit = DEFAULT_LIMIT, page = 0) {
+    try {
+      const user = await User.findById(userId);
+      const posts = await Post.aggregate([
+        {
+          $facet: {
+            posts: [
+              { $match: { user: new ObjectId(userId) } },
+              { $lookup: { from: 'tags', localField: 'tags', foreignField: '_id', as: 'tags' } },
+              { $skip: page > 0 ? (page - 1) * limit : 0 },
+              { $limit: limit },
+            ],
```

Comment:
```suggestion
              { $match: { user: new ObjectId(userId) } }, 
              { $skip: page > 0 ? (page - 1) * limit : 0 },
              { $limit: limit },
              { $lookup: { from: 'tags', localField: 'tags', foreignField: '_id', as: 'tags' } },
            ],
```
Optimize it. Move it after 'match'